components/central-application-gateway/internal/proxy/factory_test.go

```diff
@@ -39,21 +39,24 @@ func TestProxyFactory(t *testing.T) {
 		ProxyCacheTTL: 10,
 	}
 
-	createTestServer := func(path string) *httptest.Server {
+	createTestServer := func(t *testing.T, path string) *httptest.Server {
 		return NewTestServer(func(req *http.Request) {
-			assert.Equal(t, req.Method, http.MethodGet)
-			assert.Equal(t, req.RequestURI, path)
+			assert.Equal(t, http.MethodGet, req.Method)
+			assert.Equal(t, path, req.URL.String())
 		})
 	}
 
-	createMockServiceDeffService := func(apiIdentifier model.APIIdentifier, targetURL string, credentials *authorization.Credentials) metadatamocks.ServiceDefinitionService {
-		serviceDefServiceMock := metadatamocks.ServiceDefinitionService{}
-		serviceDefServiceMock.On("GetAPIByServiceName", apiIdentifier.Application, apiIdentifier.Service).Return(&metadatamodel.API{
-			TargetUrl:   targetURL,
-			Credentials: credentials,
-		}, nil).Once()
-
-		return serviceDefServiceMock
+	createMockServiceDeffService := func(encodeURL bool) func(apiIdentifier model.APIIdentifier, targetURL string, credentials *authorization.Credentials) metadatamocks.ServiceDefinitionService {
```

Comment:
What an abomination :'D Is there a more readable way to do it? I mean passing `createMockServiceDeffService(true)` as a function could be misleading at a first glance. And there is no real indication of what this `true` is unless you read this

Reply:
<img width="386" alt="image" src="https://user-images.githubusercontent.com/9433601/213993956-caddb7f3-f498-4544-bb5c-fe8f5d052837.png">
In an IDE with typehints it seems quite readable, but I can change it

Reply:
Probably yes, but in an IDE without it, it's quite mysterious